cli/src/crypto.js

```diff
@@ -1,18 +1,56 @@
 const crypto = require('crypto');
 
-const encrypt = (value, passphrase, algorithm = 'aes-256-ctr') => {
-    const cipher = crypto.createCipher(algorithm, passphrase);
-    const crypted = cipher.update(value, 'utf-8', 'hex');
+const ALGORITHM = 'aes-256-ctr';
+const KEY_BYTE_LENGTH = 32;
+const IV_LENGTH = 16;
 
-    return `${algorithm}:${crypted + cipher.final('hex')}`;
+const hexToBuffer = hex => new Buffer(hex, 'hex');
+const bufferToHex = buffer => buffer.toString('hex');
+
+const bufferize = (key, tryToCast = true) => {
+    if (!Buffer.isBuffer(key) || key.length !== KEY_BYTE_LENGTH) {
+        if (tryToCast) {
+            return bufferize(hexToBuffer(key), false);
+        }
+
+        throw new Error(`The "key" parameter should be a ${KEY_BYTE_LENGTH} bytes Buffer`);
+    }
+
+    return key;
+};
+
+const encrypt = (value, hexKey) => {
+    const key = bufferize(hexKey);
+    const iv = crypto.randomBytes(IV_LENGTH);
+
+    const cipher = crypto.createCipheriv(ALGORITHM, key, iv);
+    const cipherText = cipher.update(value, 'utf-8', 'hex') + cipher.final('hex');
+
+    return `${ALGORITHM}:${iv.toString('hex')}:${cipherText}`;
 };
 
-const decrypt = (entry, passphrase) => {
-    const [algorithm, value] = entry.split(':');
-    const decipher = crypto.createDecipher(algorithm, passphrase);
-    const decrypted = decipher.update(value, 'hex', 'utf-8');
+const decrypt = (entry, hexKey) => {
+    const key = bufferize(hexKey);
+
+    const [algorithm, hexIV, value] = entry.split(':');
+
+    if (algorithm !== ALGORITHM) {
+        throw new Error(`Unsupported algorithm: ${algorithm}`);
+    }
 
-    return decrypted + decipher.final('utf-8');
+    const iv = hexToBuffer(hexIV);
+    const decipher = crypto.createDecipheriv(ALGORITHM, key, iv);
+
+    return decipher.update(value, 'hex', 'utf-8') + decipher.final('utf-8');
```

Comment:
There might be an issue with characters encoded on 2 bytes just between the initial cipher and the final.
By instance, have a string of 257 characters ending with a character encoded on 2 bytes, what happens ?